internal/gatewayapi/testdata/securitypolicy-with-extauth-invalid-no-matching-port.in.yaml

```diff
@@ -0,0 +1,64 @@
+gateways:
+  - apiVersion: gateway.networking.k8s.io/v1
+    kind: Gateway
+    metadata:
+      namespace: default
+      name: gateway-1
+    spec:
+      gatewayClassName: envoy-gateway-class
+      listeners:
+        - name: http
+          protocol: HTTP
+          port: 80
+          allowedRoutes:
+            namespaces:
+              from: All
+httpRoutes:
+  - apiVersion: gateway.networking.k8s.io/v1
+    kind: HTTPRoute
+    metadata:
+      namespace: default
+      name: httproute-1
+    spec:
+      hostnames:
+        - www.foo.com
+      parentRefs:
+        - namespace: default
+          name: gateway-1
+          sectionName: http
+      rules:
+        - matches:
+            - path:
+                value: /foo
+          backendRefs:
+            - name: service-1
+              port: 8080
+services:
+  - apiVersion: v1
+    kind: Service
+    metadata:
+      namespace: default
+      name: http-backend
+    spec:
+      ports:
+        - port: 8080
+securityPolicies:
+  - apiVersion: gateway.envoyproxy.io/v1alpha1
+    kind: SecurityPolicy
+    metadata:
+      namespace: default
+      name: policy-for-gateway
+    spec:
+      targetRef:
+        group: gateway.networking.k8s.io
+        kind: Gateway
+        name: gateway-1
+        namespace: default
+      extAuth:
+        http:
+          Name: http-backend
```

Comment:
shouldnt this be ?
```
http:
  backendRef:
    namespace: ...
    port: 80
```

Reply:
BackendObjectReference is an incline struct in the API.

```
type GRPCExtAuthService struct {
	gwapiv1.BackendObjectReference `json:",inline"`
}

type HTTPExtAuthService struct {
	gwapiv1.BackendObjectReference `json:",inline"`

	Path *string `json:"path,omitempty"`

	HeadersToBackend []string `json:"headersToBackend,omitempty"`
}
```

Or we can make it explicit like this:

```
type GRPCExtAuthService struct {
	BackendRef gwapiv1.BackendObjectReference 
}

type HTTPExtAuthService struct {
	BackendRef gwapiv1.BackendObjectReference

	Path *string `json:"path,omitempty"`

	HeadersToBackend []string `json:"headersToBackend,omitempty"`
}
```